CNV-29048: Release note: CHANGE (metrics changes)

aspauldi opened the pull request:
Version(s): 4.14

Issue: [CNV-29048](https://issues.redhat.com/browse/CNV-29048)

Link to docs preview: [New and changed features](https://64674--docspreview.netlify.app/openshift-enterprise/latest/virt/release_notes/virt-4-14-release-notes#virt-4-14-new) 


QE review:
- [x] QE has approved this change.

Additional information:

aspauldi commented:
@sradco , I've created the release note. The link to the article won't work until it is published, but can you please review the note text? Thanks.

aspauldi commented:
@sradco, Can you take a look at the release note? Thanks!

sradco commented on virt/release_notes/virt-4-14-release-notes.adoc:
```diff
@@ -80,6 +80,9 @@ The SVVP Certification applies to:
 //CNV-28732 Release note: NEW
 
+//CNV-29048 Release note: NEW Metrics name changes
+* Many OpenShift Virtualization metrics have changed. These changes could affect your custom dashboards. See link:https://access.redhat.com/articles/7028805[OpenShift Virtualization 4.14 metric changes] for details.
```
Hi @aspauldi,  Please comment that the link includes both metrics that changed name and deprecated alerts that would change in 4.15.

aspauldi replied:
Hi @sradco, do  you mean that there are metrics whose names have changed in 4.14 and also alerts (not metrics) that are being deprecated in 4.14 and changed (but not removed) in 4.15? 

When I look [here](https://docs.google.com/spreadsheets/d/1buMYpxGK29VzgonbZaA4uimcJ02q5YrDVy80QQ5mulI/edit#gid=559835819), I see only a list of metric name changes in 4.14.

sradco replied:
Sorry, it's all about metrics. I made a mistake in the comment and updated it now

aspauldi replied:
How would metrics deprecated in 4.14 change in 4.15? Would the metrics still exist in 4.15 (removal would happen at some point in the future, after 4.15) but be renamed?

aspauldi commented:
@kmajcher-rh, would you be able to look at this PR (in lieu of Shirly)? Thanks

kmajcher-rh commented:
@aspauldi - i just have one question about "The following metrics are deprecated in 4.14 and will be changed in 4.15." section, which is followed by a table with list of metrics with old and new name. I assume new name will be effective in 4.15 only, and old name still works in 4.14? Is this how a reader should understand this?

aspauldi commented:
Hi @kmajcher-rh, I was confused about that too, what is meant by "The following metrics are deprecated in 4.14 and will be changed in 4.15." I had asked Shirly about that in a comment last week (see comment ^) and was awaiting her response.  Perhaps we need to wait for Shirly to review?

ctomasko commented:
@aspauldi Please make sure to reference the correct BZ https://bugzilla.redhat.com/show_bug.cgi?id=2179660 / Jira https://issues.redhat.com/browse/CNV-27054 issue number.   Please note that CNV-29048 is a tracker issue for the release note and the bug needs to reference the original BZ 2179660

aspauldi commented:
/label cnv

aspauldi commented:
@sradco, I've updated the note, with your new detail. You can find it in the New and changed features section of the release notes.

@dbasunag, can you review the release note? 

Thanks.

ctomasko reviewed (commented):
@aspauldi Unfortunately, the link to https://access.redhat.com/articles/7028805 is a private link for internal Red Hat only, so it won't benefit our customers. 
I recommend including both of the tables in the article (4.14 changed metrics and 4.14 deprecated metrics) in the release notes.  @apinnick do you agree that we should include these tables since the link to the article is not accessible by customers?

dbasunag commented on virt/release_notes/virt-4-14-release-notes.adoc:
```diff
@@ -85,6 +85,8 @@ The SVVP Certification applies to:
 //CNV-28729 Release note: New
 * You can use a xref:../../virt/virtual_machines/advanced_vm_management/virt-schedule-vms.adoc#virt-vm-custom-scheduler_virt-schedule-vms[custom scheduler] to schedule a virtual machine (VM) on a node.
+//CNV-29048 Release note: NEW Metrics name changes
+* Many OpenShift Virtualization metrics have changed or will change in a future version. These changes could affect your custom dashboards. See link:https://access.redhat.com/articles/7028805[OpenShift Virtualization 4.14 metric changes] for details.
```
@aspauldi the linked document has "PRIVATE: This document is for internal use and is not accessible to customers." header, is it appropriate for release note, that would be out for public consumption?

@sradco https://access.redhat.com/articles/7028805 includes metrics that would be renamed in 4.15. Referencing it without clearly mentioning which are getting updated in 4.14 vs 4.15, would be more confusing I feel.

apinnick replied:
@dbasunag The linked article has not yet been published.

@sradco Is it ready for publication?

dbasunag replied:
@apinnick I see the article https://access.redhat.com/articles/7028805 is published, but some of the metrics that are mentioned there did not get renamed till 4.15. This would be misleading. @sradco what do you suggest we do?

aspauldi commented:
Hi @dbasunag, the article is published. Can you please review? Thanks!

ctomasko commented:
@sradco   We would like you to review and clarify the wording in the article to be clear about what changed in 4.14. https://access.redhat.com/articles/7028805 

The article must be clear about:

1. the metrics that changed names in 4.14
2. the metrics that are deprecated in 4.14

If there are metrics that will change in 4.15, please list them separately, or don't mention them at all since 4.15 is not yet released.    Thank you

sradco commented:
We need to notify our customers on coming changes in 4.15 so that they can plan to update their metrics collection pipelines in case they use an external metrics store. This is the reason we also list the metrics that are deprecated in 4.14 and in 4.15 will have a new name.

aspauldi commented:
/label peer-review-needed

mletalie commented:
/remove-label peer-review-needed
/remove-label peer-review-in-progress
/label peer-review-done
/lgtm

aspauldi commented:
/label merge-review-needed

mletalie commented:
/label peer-review-done

mletalie commented:
/remove-label peer-review-needed
/remove-label peer-review-in-progress
